src/aws-cpp-sdk-core/include/aws/core/client/RequestCompression.h

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#pragma once
+
+#include <aws/core/Core_EXPORTS.h>
+#include <aws/core/client/ClientConfiguration.h>
+#include <aws/core/utils/Outcome.h>
+#include <aws/core/utils/memory/stl/AWSStreamFwd.h>
+#include <aws/core/utils/memory/stl/AWSString.h>
+#include <aws/core/utils/memory/stl/AWSVector.h>
+
+namespace Aws {
+namespace Client {
+
+enum class CompressionAlgorithm { NONE, GZIP };
+
+Aws::String GetCompressionAlgorithmId(const CompressionAlgorithm &algorithm);
+
+class AWS_CORE_API RequestCompression {
```

Comment:
it does not define destructor as virtual, thus mark this class as final

Reply:
done with final